jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/XGBoost.java

```diff
@@ -196,16 +196,11 @@ public static Booster train(
         for (int i = 0; i < metricsOut.length; i++) {
           metrics[i][iter] = metricsOut[i];
         }
-
-        boolean decreasing = true;
-        float[] criterion = metrics[metrics.length - 1];
-        for (int shift = 0; shift < Math.min(iter, earlyStoppingRound) - 1; shift++) {
-          decreasing &= criterion[iter - shift] <= criterion[iter - shift - 1];
-        }
-
-        if (!decreasing) {
+        boolean onTrack = judgeIfTrainingOnTrack(params, earlyStoppingRounds, metrics, iter);
```

Comment:
Can we put this check inside of ```earlyStoppingRounds > 0```? As if users don't  set early stopping, we don't need to pay effort to check it on track or not.